src/FFmpeg.cpp

```diff
@@ -302,22 +307,53 @@ BEGIN_EVENT_TABLE(FFmpegNotFoundDialog, wxDialogWrapper)
    EVT_BUTTON(wxID_OK, FFmpegNotFoundDialog::OnOk)
 END_EVENT_TABLE()
 
+struct SafeAVFormatPathUpdater final
+{
+   explicit SafeAVFormatPathUpdater(const wxString& newPath)
+   {
+      mHasOldValue = AVFormatPath.Read(&mOldValue);
+
+      AVFormatPath.Write(newPath);
+   }
+
+   ~SafeAVFormatPathUpdater()
+   {
+      if (mRevertToOld)   
+      {
+         if (mHasOldValue)
+            AVFormatPath.Write(mOldValue);
+         else
+            AVFormatPath.Delete();
+      }
+
+      gPrefs->Flush();
+   }
+
+   void CommitValue() noexcept
+   {
+      mRevertToOld = false;
+   }
+
+   wxString mOldValue;
+   bool mHasOldValue;
+   bool mRevertToOld { false };
```

Comment:
I see what you are doing, but mRevertToOld must be constructed as true by default.

Should the destructor catch exceptions?

Reply:
Maybe this class wants to be generalized as a utility in Prefs.h.  It would want a template parameter for the Settings object.  But, another day.

Reply:
>I see what you are doing, but mRevertToOld must be constructed as true by default.

My bad

>Should the destructor catch exceptions?

I'm not sure why and which? Is there something good we can do?

Reply:
Maybe this class wants to be generalized as a utility in Prefs.h

We definitely should